FBRetainCycleDetector/Layout/Classes/FBClassStrongLayout.mm

```diff
@@ -10,37 +10,43 @@
 #import "FBClassStrongLayout.h"
 
 #import <math.h>
+#import <memory>
 #import <objc/runtime.h>
+#import <vector>
 
 #import <UIKit/UIKit.h>
 
 #import "FBIvarReference.h"
 #import "FBObjectInStructReference.h"
 #import "FBStructEncodingParser.h"
+#import "Struct.h"
+#import "Type.h"
 
 /**
  If we stumble upon a struct, we need to go through it and check if it doesn't retain some objects.
  */
-static NSArray *FBGetReferencesForObjectsInStructEncoding(FBIvarReference *ivar, NSString *encoding) {
+static NSArray *FBGetReferencesForObjectsInStructEncoding(FBIvarReference *ivar, std::string encoding) {
   NSMutableArray<FBObjectInStructReference *> *references = [NSMutableArray new];
 
-  FBParsedStruct *parsedStruct = FBParseStructEncodingWithName(encoding,
-                                                         ivar.name);
-  NSArray<FBParsedType *> *types = [parsedStruct flattenTypes];
-
+  std::string ivarName = std::string([ivar.name cStringUsingEncoding:NSUTF8StringEncoding]);
+  FB::RetainCycleDetector::Parser::Struct parsedStruct =
```

Comment:
Consider using `auto` as the type here, and many other places. Any time the same type appears when declaring a variable and then again to construct or initialize the variable, consider making the variable have type `auto`.